jina/executors/indexers/vector.py

```diff
@@ -306,3 +310,9 @@ def _cdist(self, *args, **kwargs):
             return cdist(*args, **kwargs, metric=self.metric)
         except ModuleNotFoundError:
             raise ModuleNotFoundError(f'your metric {self.metric} requires scipy, but scipy is not found')
+
+    def _is_empty(self):
```

Comment:
I think the emptiness is extracted from the `_size` and not from the `query_handler` being None

Reply:
Got it. The problem was that we were passing keys even though there wasn't any data. Fixed that now